src/Transport/AzureServiceBusTransport.cs

```diff
@@ -333,5 +333,10 @@ public IWebProxy WebProxy
 
         internal string FullyQualifiedNamespace { get; set; }
         internal TokenCredential TokenCredential { get; set; }
+
+        /// <summary>
+        /// Register call that allows to customize the native ServiceBusMessage just before it is dispatched to the Azure Service Bus SDK client
+        /// </summary>
+        public NativeMessageCustomizerCallback NativeMessageCustomization { get; set; }
```

Comment:
Could we name this containing the word "outgoing" or anything the like to make it clear what the purpose is? Because technically, we could also have an incoming customizer.

Also what is the benefit of having a custom delegate over Action delegate? Is it the ability to name things? Could we maybe also promote this approach if we stick to it to our design principles in platform? I'm aware we have done the same for the OnMessage and OnError delegates which seem to be special cases due to the complexity and low level embedding into the transport. Given this is user facing it seems applying the framework guidelines would be a sensible thing to do in my opinion

>  DO use the new Func<...>, Action<...>, or Expression<...> types instead of custom delegates, when defining APIs with callbacks.

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/events-and-callbacks

Reply:
To echo the name aspect, "customizer" and "callback" together read somewhat redundant. As a user I'd understand it's a hook to manipulate the dispatched message.